alica_engine/src/engine/RuleBook.cpp

```diff
@@ -31,17 +32,18 @@ using std::endl;
 /**
  * Basic constructor
  */
-RuleBook::RuleBook(AlicaEngine* ae, PlanBase* pb)
-        : _tm(ae->getTeamManager())
-        , _ps(new PlanSelector(ae, pb))
-        , _log(ae->editLog())
+RuleBook::RuleBook(ConfigChangeListener& configChangeListener, Logger& log, SyncModule& syncModule, TeamObserver& teamObserver, const TeamManager& teamManager,
+        const PlanRepository& planRepository, PlanBase* pb)
+        : _logger(log)
+        , _syncModule(syncModule)
+        , _teamManager(teamManager)
         , _pb(pb)
-        , _sm(ae->editSyncModul())
+        , _ps(new PlanSelector(teamObserver, teamManager, pb))
```

Comment:
Is there a reason for the PlanSelector to be a raw ptr? Also I couldnt find a delete anywhere in the code, it seems like the memory allocated for the PlanSelector is never deallocated.

Reply:
it is a unique_ptr. Better to use make_unique instead of new